core/domain/exp_jobs_one_off.py

```diff
@@ -528,6 +528,123 @@ def reduce(key, values):
 
 
+class ExplorationMathRichTextInfoModelGenerationOneOffJob(
+        jobs.BaseMapReduceOneOffJobManager):
+    """Job that finds all the explorations with math rich text components and
+    creates a temporary storage model with all the information required for
+    generating math rich text component SVG images.
+    """
+
+    # A constant that will be yielded as a key by this job in the map function,
+    # When it finds an exploration with math rich text components without SVGs.
+    _SUCCESS_KEY = 'exploration-with-math-tags'
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+
+        exploration = exp_fetchers.get_exploration_from_model(item)
+        try:
+            exploration.validate()
+        except Exception as e:
+            logging.error(
+                'Exploration %s failed non-strict validation: %s' %
+                (item.id, e))
+            yield (
+                'validation_error',
+                'Exploration %s failed non-strict validation: %s' %
+                (item.id, e))
+            return
+        html_strings_in_exploration = ''
+        for state in exploration.states.values():
+            html_strings_in_exploration += (
+                ''.join(state.get_all_html_content_strings()))
+        list_of_latex_values_without_svg = (
+            html_validation_service.
+            get_latext_values_without_svg_from_html(
+                html_strings_in_exploration))
+        if len(list_of_latex_values_without_svg) > 0:
+            yield (
+                ExplorationMathRichTextInfoModelGenerationOneOffJob.
+                _SUCCESS_KEY,
+                (item.id, list_of_latex_values_without_svg))
+
+    @staticmethod
+    def reduce(key, values):
+        if key == (
+                ExplorationMathRichTextInfoModelGenerationOneOffJob.
+                _SUCCESS_KEY):
+            final_values = [ast.literal_eval(value) for value in values]
+            estimated_no_of_batches = 1
+            approx_size_of_math_svgs_bytes_in_a_batch = 0
+            exploration_math_rich_text_info_list = []
+            longest_raw_latex_string = ''
+            total_number_of_svgs_required = 0
+            for exp_id, list_of_latex_values_without_svg in final_values:
+                math_rich_text_info = (
+                    exp_domain.ExplorationMathRichTextInfo(
+                        exp_id, True, list_of_latex_values_without_svg))
+                exploration_math_rich_text_info_list.append(
+                    math_rich_text_info)
+
+                approx_size_of_math_svgs_bytes = (
+                    math_rich_text_info.get_svg_size_in_bytes())
+                total_number_of_svgs_required += (
+                    len(
+                        list_of_latex_values_without_svg))
+                longest_raw_latex_string = (
+                    max(
```

Comment:
Can move to previous line and drop outer paren

Reply:
done